### ISSUE TITLE
20.sh issue on debian-10 

### ISSUE THREAD

**Issue by dkarnikis** [binpash/pash#466]
When running `20.sh` script from unix50 with the following flags:
`export PASH_FLAGS='-w16 --r_split --dgsh_tee --r_split_batch_size 1000000'`
I get:
`r_wrap: There is a problem with reading the block: Success`
However, by reducing the size to 1000, the issue seems to be fixed.
I encountered this issue when running on the docker version of pash on debian on branch setup-fixes.

To reproduce it do the following:
1. docker run -it binpash/pash:debian-10 /bin/bash (haven't checked if this applies to native instance as well)
2. build pash on branch main
3. checkout to setup-fixes (so you can run with the small input)
4. source run.par.sh (evaluation/benchmarks/)
5. export PASH_FLAGS='-w16 --r_split --dgsh_tee --r_split_batch_size 1000000'
6. unix50_pash --small (so you can use the small dataset)

I think the issue is related to the small input size of the data (--small)
@tammam1998 could you help me with that?

**Comment by tammam1998** [comment]
Does the issue happen only on debian:10, what about ubuntu?

**Comment by dkarnikis** [comment]
I only tested debian:10 (since it matches the OS running on deathstar). However, I could not reproduce the issue all the times I tested the experiment (I think you had encountered a similar issue in the past).

**Comment by tammam1998** [comment]
Yes, I thought I fixed it with a pull request couple weeks ago. Did you not recompile the runtime after merging the future branch by any chance?

**Comment by dkarnikis** [comment]
I tested it on a fresh installation/setup on future

**Comment by tammam1998** [comment]
I see, I will look into it

**Comment by tammam1998** [comment]
Hi @dkarnikis,

I followed the same setup as you and I'm having some problems.

1- I can't switch to other git branches, do I have to run any commands to get the full repo? Also, the main branch was a little outdated fixed by running git pull.

2- I tried running test_evalutation scripts (on the latest main branch) and only 26/54 test cases pass. This is deterministic and the results are the same even with no pash flags like r_split and dgsh_tee. 

3- In step 2 you say build on main but I believe the fix I submitted is only on future rn. When I followed the steps (outside of docker on a native machine) I got the same error. However, after I ran `make all` on the setup-fixes branch the problem stopped happening.
